lib/fs/fsfileinfo_windows.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2017 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package fs
+
+import (
+	"os"
+)
+
+func (e fsFileInfo) Mode() FileMode {
+	m := e.FileInfo.Mode()
+	if e.Size() > 0 {
+		// "symlinks" with size are in fact "hard" links, such as ntfs deduped files
+		m &^= os.ModeSymlink
+	}
+	return FileMode(m)
+}
+
+func (e fsFileInfo) IsRegular() bool {
+	if e.Size() > 0 {
+		return true
+	}
+	return e.FileInfo.Mode().IsRegular()
+}
+
+func (e fsFileInfo) IsSymlink() bool {
+	if e.Size() > 0 {
+		return false
+	}
+	return e.FileInfo.Mode()&os.ModeSymlink == os.ModeSymlink
```

Comment:
Do you even need this? Can't we just call e.Mode() instead of e.FileInfo.Mode()?